app/stac_api/collection_temporal_extent.py

```diff
@@ -318,6 +384,9 @@ def update_end_temporal_extent_on_item_delete(collection, old_end_datetime, item
             the id of the item being treated (pk)
         qs: QuerySet | None
             queryset with all items of the collection excluding the one being updated. (optional)
+        qs_other_items_with_properties_datetimes: QuerySet | None
+            Already evaluated queryset with all items (but the one currently updated) that have
```

Comment:
See comment about exists(), then it is not fully evaluated.

Reply:
See comment above when updating on item_update